seqr/views/apis/individual_api.py

```diff
@@ -118,42 +114,101 @@ def edit_individuals_handler(request, project_guid):
+                                errors.append(
+                                    "{} is recorded as the {} of {} but they have different family ids: {} and {}".format(
+                                        indiv_id, parent_id_type, child.individual_id, family_id, child.family.family_id)
+                                )
+
+            # Validate parents have correct sex and family
+            if update[parent_id_key] and (update[parent_id_key] != getattr(model, parent_id_attr) or family_id != model.family.family_id):
+                parent = Individual.objects.filter(individual_id=update[parent_id_key])
+                if len(parent):
+                    parent = parent[0]
+                    if update_individuals.get(parent.guid):
+                        sex = update_individuals[parent.guid][1]['sex']
+                        parent_family_id = update_individuals[parent.guid][1]['family']['familyId']
+                    else:
+                        sex = parent.sex
+                        parent_family_id = parent.family.family_id
+
+                    if sex != expected_sex:
+                        errors.append("{} is recorded as {} and also as the {} of {}".format(update[parent_id_key], sex, parent_id_type, indiv_id))
+                    if first_check and family_id != parent_family_id:
+                        errors.append("{} is recorded as the {} of {} but they have different family ids: {} and {}".format(
+                            update[parent_id_key], parent_id_type, indiv_id, parent_family_id, family_id)
+                        )
+                else:
+                    errors.append("{} is recorded as the {} of {} but does not exist".format(update[parent_id_key], parent_id_type, indiv_id))
 
-    # TODO more validation
-    errors, warnings = validate_fam_file_records(modified_individuals_list)
     if errors:
-        return create_json_response({'errors': errors, 'warnings': warnings})
+        return create_json_response({'errors': errors}, status=400, reason='Invalid updates')
 
-    updated_families, updated_individuals = add_or_update_individuals_and_families(
-        project, modified_individuals_list)
+    individuals_by_guid = {}
+    families = []
+    for individual_guid, (model, update) in update_individuals.items():
+        family_id = update['family']['familyId']
 
-    individuals_by_guid = {
-        individual.guid: _get_json_for_individual(individual, request.user) for individual in updated_individuals
-    }
+        family, created = Family.objects.get_or_create(
+            project=project,
+            family_id=family_id)
+
+        if created:
+            logger.info("Created family: %s", family)
+            if not family.display_name:
+                family.display_name = family.family_id
+                family.save()
+
+        families.append(family)
+
+        if family_id != model.family.family_id:
```

Comment:
when would this happen?

Reply:
If you updated the `family_id` in the bulk update for individuals